### ISSUE TITLE
[0.5.10]  new "Create steam shortcut" option isnt there on some installs

### ISSUE THREAD

**Issue by BaronKrause** [lutris/lutris#4210]
### Bug description

I have tested this on 2 PC's and the new "Create steam shortcut" option doesn't show up on the system that Installed Steam using Pop OS's repo Steam installer. Steam gets installed into "/home/user/.steam/debian-installation" instead of "/home/user/.local/share/Steam" like the generic website installer uses. The test machine using that install method shows the new option along with the Desktop and Application shortcut options as expected.

In both cases Lutris can see your steam games under the sources section and can launch Steam from the runners section.

Both are also using the Steam Beta update branch selected inside Steam Preferences.

### How to Reproduce

1. Install Steam on Pop OS (and I assume Ubuntu?) and notice that it goes to ~/.steam/debian-installer
2. Open Lutris
3. Start a Game install from a Lutris Source
4. The new "Create steam shortcut" will not be there.

### Expected behavior

The "Create steam shortcut" option to show like it does on the Pop OS system with the generic Steam installer.

### Log output

```shell
No log error to show, the option just isn't made available in the example debian-installation install.
```

### System Information

```shell
[System]
OS:              Pop!_OS 21.10 impish
Arch:            x86_64
Kernel:          5.16.15-76051615-generic
Desktop:         pop:GNOME
Display Server:  x11

[CPU]
Vendor:          GenuineIntel
Model:           11th Gen Intel(R) Core(TM) i7-11850H @ 2.50GHz
Physical cores:  8
Logical cores:   16

[Memory]
RAM:             15.3 GB
Swap:            4.0 GB

[Graphics]
Vendor:          NVIDIA Corporation
OpenGL Renderer: NVIDIA GeForce RTX 3070 Laptop GPU/PCIe/SSE2
OpenGL Version:  4.6.0 NVIDIA 510.54
OpenGL Core:     4.6.0 NVIDIA 510.54
OpenGL ES:       OpenGL ES 3.2 NVIDIA 510.54
Vulkan:          Supported
```

### Checklist:

- [X] I'm not asking for support with a game or the wine runner.
- [X] I have followed the above mentioned guides and have all the graphics and wine dependencies installed.
- [X] I have checked for existing issues that describe my problem prior to opening this one.
- [X] I understand that improperly formatted bug reports may be closed without explanation.

**Comment by liberavia** [comment]
Can you confirm that the game you wanted to link to steam is a non-steam game? The feature is only available for non-steam or winesteam games. Which is the game you wanted to link?

**Comment by strycore** [comment]
I've move "~/.steam/debian-installation" to be the 1st element in the list, check if that makes any difference.

**Comment by liberavia** [comment]
I still would like to know which game with which runner you did try to link to steam. As said, the feature just adds games to steam which don't have any steam context.

**Comment by liberavia** [comment]
> I've moved "~/.steam/debian-installation" to be the 1st element in the list, check if that makes any difference.

I don't think this should be a problem here due it checks all pathes equally as you can see here:
https://github.com/lutris/lutris/blob/a518659349598f4219575a0dafacc06fb12bc439/lutris/util/steam/config.py#L42-L50

The result of all pathes will be processed here:
https://github.com/lutris/lutris/blob/a518659349598f4219575a0dafacc06fb12bc439/lutris/util/steam/shortcut.py#L12-L15

Not having the option shown has to be that one of these conditions are failing:
https://github.com/lutris/lutris/blob/master/lutris/game_actions.py#L138-L143

**Comment by liberavia** [comment]
> The test game was Overwatch from the current Lutris hosted install script. But I’ve also tried a few others, all with the same results.
> 
> The machines were 3 Pop!_OS machines, 2 with the repo installed steam (with the Debian-installation path) and 1 Steam website downloaded deb installer. Only the Website installed one gets the new third option to make the link in steam.
> 
> To test that change, do I need to compile from source or replace a config file? In all 3 cases Lutris was installed from the Pop!_OS repos as recommended on the Lutris website and seems to be kept up to date almost immediately.

Ok, so we can sort the steam-game assumption out. If I'd guess I would say that yet you don't have a single shortcut existing inside steam, right?

If true, could you please try to add a random non-steam game (or program) to steam the usual way inside steam and afterwards trying to restart lutris and give linking of overwatch another try? Thx

**Comment by BaronKrause** [comment]
> > The test game was Overwatch from the current Lutris hosted install script. But I’ve also tried a few others, all with the same results.
> > The machines were 3 Pop!_OS machines, 2 with the repo installed steam (with the Debian-installation path) and 1 Steam website downloaded deb installer. Only the Website installed one gets the new third option to make the link in steam.
> > To test that change, do I need to compile from source or replace a config file? In all 3 cases Lutris was installed from the Pop!_OS repos as recommended on the Lutris website and seems to be kept up to date almost immediately.
> 
> Ok, so we can sort the steam-game assumption out. If I'd guess I would say that yet you don't have a single shortcut existing inside steam, right?
> 
> If true, could you please try to add a random non-steam game (or program) to steam the usual way inside steam and afterwards trying to restart lutris and give linking of overwatch another try? Thx
> 
> Edit: I'll suspect that there is no shortcut.vdf file yet and therefore this condition will fail and therefore the option is not offered:
> 
> https://github.com/lutris/lutris/blob/4a174777a0b5131224002277876624a126da87e1/lutris/game_actions.py#L140

That did fix it on one of the machines I tested with, I just added some default system program to steam and all of a sudden Lutris now offers that new steam link option. Seems to continue working after removing it as well.

**Comment by liberavia** [comment]
Thank you for confirming.

So this issue doesn't need a triage IMHO. 

I'm currently not sure how to solve the issue from Lutris' point of view. I have to check if creating a file is an option here or if we just live with the fact that we just modify an existing file. I'll do some investigation and will eventually ad a PR. 

In the meantime suggestions/opinions are of course welcome

**Comment by strycore** [comment]
We're already modifying the existing file so I don't see a problem creating it if it doesn't exist already. I have a shortcuts.vdf file that is empty and it appears to contain nothing other than the key 'shortcuts' which means it's totally fine if we create it ourselves.

![Screenshot from 2022-04-14 19-25-39](https://user-images.githubusercontent.com/280336/163512787-fd441ecc-121f-43be-a3c5-35b2e57495ca.png)

**Comment by strycore** [comment]
> I didn't get to test the latest version of the cloned repo, but the latest changelog indicates it may have been fixed for 0.5.11?

Yes, this appears to be fixed.
